src/word_search_generator/__init__.py

```diff
@@ -76,39 +79,35 @@ def json(self) -> str:
         )
 
     @property
-    def level(self) -> int:
-        """The difficulty level of the puzzle."""
-        return self._level
+    def valid_directions(self) -> DirectionSet:
+        return self._valid_directions
 
-    @level.setter
-    def level(self, val: int):
-        """Set the difficulty level of the word search.
-
-        Level 1 (Easy): Words can go forward in directions
-        EAST (E), or SOUTH (S).
-        Puzzle size is small by default.
+    @valid_directions.setter
+    def valid_directions(self, val):
```

Comment:
This could be hard to type if you plan to accept direction objects (which I haven't looked at yet) here as well. For now it could be typed as `def valid_directions(self, val: int | str | Iterable[str]): -> DirectionSet:`.